### PR TITLE
Move all buildflags to build_config header

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/377271365

### CODE REVIEW

File: cobalt/build/configs/BUILD.gn [hunk 1]
```diff
@@ -32,3 +37,22 @@ config("default_includes") {
...
   }
 }
+
+config("buildflag_defines") {
+  defines = []
+  if (is_cobalt) {
+    defines += [ "ENABLE_BUILDFLAG_IS_COBALT" ]
+  }
+  if (is_starboard) {
+    defines += [ "OS_STARBOARD" ]
```
Comment (hlwarriner) [inline cobalt/build/configs/BUILD.gn:47]: Can we get rid of `OS_STARBOARD` and just have a `use_starboard` variable that defines `BUILDFLAG_INTERNAL_USE_STARBOARD`?

File: cobalt/build/configs/initialize_variables.gni [hunk 2]
```diff
@@ -34,9 +34,6 @@ declare_args() {
   build_with_separate_cobalt_toolchain = target_os != "android"
 }
 
-assert(!(is_starboard && is_native_target_build),
```
Comment (hlwarriner) [inline cobalt/build/configs/initialize_variables.gni:37]: Just wondering why this is removed?

File: cobalt/build/configs/BUILD.gn (hunk 1 above)
Comment (andrewsavage1) [reply cobalt/build/configs/BUILD.gn:47]: We can, my idea here was to mimic the other platform macros in //build/build_config.h (i.e. OS_CHROMEOS). But of course that makes it inconsistent with the rest of the defines here, which looks pretty strange. I'm fine with changing it, but maybe take a look a the other OS_FOO defines in build_config.h and see if you still would prefer to?

Comment (andrewsavage1) [reply cobalt/build/configs/BUILD.gn:47]: See https://github.com/youtube/cobalt/blob/232171bbaefb3afb59b1a47abde13831d38e2612/build/config/linux/BUILD.gn#L40 and https://github.com/youtube/cobalt/blob/232171bbaefb3afb59b1a47abde13831d38e2612/build/build_config.h#L79

File: cobalt/build/configs/initialize_variables.gni (hunk 2 above)
Comment (andrewsavage1) [reply cobalt/build/configs/initialize_variables.gni:37]: I misread the conditional in here, I'll add it back

File: cobalt/build/configs/BUILD.gn (hunk 1 above)
Comment (hlwarriner) [reply cobalt/build/configs/BUILD.gn:47]: Ah yeah I see. I'm ok with keeping it.

**Review by hlwarriner — CHANGES_REQUESTED** [review]
(Just for the `is_native_target_build` conditional)

**Review by hlwarriner — APPROVED** [review]

**Review by kaidokert — APPROVED** [review]

**Pull request merged**